common-npm-packages/artifacts-common/EntraWifUserServiceConnectionUtils.ts

```diff
@@ -0,0 +1,72 @@
+import path = require("path");
+import * as tl from 'azure-pipelines-task-lib/task';
+import { getSystemAccessToken } from "./webapi";
+import fetch from "node-fetch";
+
+tl.setResourcePath(path.join(__dirname, 'module.json'), true);
+
+export async function getFederatedWorkloadIdentityCredentials(serviceConnectionName: string, tenantId?: string) : Promise<string>
+{
+    try {
+        let tenant = tenantId ?? tl.getEndpointAuthorizationParameterRequired(serviceConnectionName, "TenantId");
```

Comment:
nit: do we need to handle the case where `tl.getEndpointAuthorizationParameterRequired` does not find a value for `TenantId` in the service connection? I'm not sure if that's possible based on the structure of a service connection in this particular case, but wanted to double-check before we start using the result in places.

Reply:
The underlying function getEndpointAuthorizationParameterRequired  throws if the required filed is not found. So as long as we are handling the catch correctly, it should already be covered